### PR TITLE
to-close: added long-running materialization support to wren

### PR DESCRIPTION
**Pull request by epinzur**

### CODE REVIEW

File: wren/compute/prepare.go [hunk 1]
```diff
@@ -188,12 +188,6 @@ func (m *Manager) executePrepare(ctx context.Context, owner *ent.Owner, prepareJ
 		}
 		subLogger.Debug().Interface("response", prepareRes).Msg("received prepare response")
 
-		for _, preparedFile := range prepareRes.PreparedFiles {
-			preparedFile.MetadataPath = preparedFile.MetadataPath
-			preparedFile.Path = preparedFile.Path
-			subLogger.Debug().Interface("prepared_file", preparedFile).Msg("these paths should be URIs")
-		}
-
```
Comment (epinzur) [inline wren/compute/prepare.go:196]: this code does nothing, removing it

File: wren/compute/materialization.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+import (
+	"context"
+
+	v1alpha "github.com/kaskada-ai/kaskada/gen/proto/go/kaskada/kaskada/v1alpha"
+	"github.com/kaskada-ai/kaskada/wren/customerrors"
+	"github.com/kaskada-ai/kaskada/wren/ent"
+	"github.com/kaskada-ai/kaskada/wren/ent/materialization"
+	"github.com/rs/zerolog/log"
+)
```
Comment (epinzur) [inline wren/compute/materialization.go:11]: wrappers around calling the new methods in the compute engine for managing long-lived materializations

File: wren/compute/materialization.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	statusResponse, err := computeClient.GetMaterializationStatus(ctx, statusRequest)
+	if err != nil {
+		subLogger.Error().Err(err).Msg("issue getting materialization status")
+		return nil, customerrors.NewComputeError(reMapSparrowError(ctx, err))
+	}
+
+	return statusResponse.Progress, nil
+}
+
+func (m *Manager) ReconcileMaterialzations(ctx context.Context) error {
```
Comment (epinzur) [inline wren/compute/materialization.go:82]: running this should make sure that all materializations in the DB are running in the compute engine

File: wren/ent/schema/materialization.go [hunk 4]
```diff
@@ -31,6 +31,7 @@ func (Materialization) Fields() []ent.Field {
 		field.Bytes("slice_request").GoType(&v1alpha.SliceRequest{}).Immutable(),
 		field.Bytes("analysis").GoType(&v1alpha.Analysis{}).Immutable(),
 		field.Int64("data_version_id"),
+		field.Enum("source_type").Values("unspecified", "files", "streams").Default("unspecified"),
```
Comment (epinzur) [inline wren/ent/schema/materialization.go:34]: add a materialization source_type to the db

File: wren/service/materialization.go [hunk 5]
```diff
@@ -228,15 +256,28 @@ func (s *materializationService) createMaterialization(ctx context.Context, owne
...
 	if err != nil {
 		return nil, err
 	}
 
-	subLogger.Debug().Msg("running materializations")
-	s.computeManager.RunMaterializations(ctx, owner)
+	switch sourceType {
+	case materialization.SourceTypeFiles:
+		subLogger.Debug().Msg("running materializations")
+		s.computeManager.RunMaterializations(ctx, owner)
```
Comment (epinzur) [inline wren/service/materialization.go:270]: if is file-backed materialization, run as before

File: wren/service/materialization_test.go [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+
+				response, err := materializationService.createMaterialization(context.Background(), owner, &v1alpha.CreateMaterializationRequest{
+					Materialization: &v1alpha.Materialization{
+						Expression:  "nachos",
+						Destination: pulsarDestination,
+					},
+				})
+				Expect(err).Should(HaveOccurred())
+				Expect(err.Error()).Should(Equal("cannot materialize tables from different source types"))
+				Expect(response).Should(BeNil())
```
Comment (epinzur) [inline wren/service/materialization_test.go:132]: test of protection around materializing from multiple source types

File: wren/compute/materialization.go [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+					log.Error().Err(err).Str("id", materializationID).Msg("failed to start materialization")
+				} else {
+					log.Debug().Str("id", materializationID).Msg("started materialization")
+					newRunningMaterailizations[materializationID] = struct{}{}
+				}
+			}
+		}
+	}
+
+	// find all materializations that were running in the previous iteration but don't exist anymore
```
Comment (jordanrfrazier) [inline wren/compute/materialization.go:129]: Can you elaborate on this logic? What is a previous iteration in this context

File: wren/service/materialization.go [hunk 8]
```diff
@@ -266,12 +307,20 @@ func (s *materializationService) DeleteMaterialization(ctx context.Context, requ
...
 
 func (s *materializationService) deleteMaterialization(ctx context.Context, owner *ent.Owner, request *v1alpha.DeleteMaterializationRequest) (*v1alpha.DeleteMaterializationResponse, error) {
-	materialization, err := s.materializationClient.GetMaterializationByName(ctx, owner, request.MaterializationName)
+	foundMaterialization, err := s.materializationClient.GetMaterializationByName(ctx, owner, request.MaterializationName)
 	if err != nil {
 		return nil, err
 	}
 
-	err = s.materializationClient.DeleteMaterialization(ctx, owner, materialization)
+	if foundMaterialization.SourceType == materialization.SourceTypeStreams {
```
Comment (jordanrfrazier) [inline wren/service/materialization.go:315]: I assume in the future we'll also have a way to have a long-running file backed materialization, so maybe a `todo` here

**Pull request closed without merging**